templates/cluster-template-flatcar.yaml

```diff
@@ -0,0 +1,265 @@
+          cluster-name: ${CLUSTER_NAME}
+        extraVolumes:
+        - hostPath: /etc/kubernetes/azure.json
+          mountPath: /etc/kubernetes/azure.json
+          name: cloud-config
+          readOnly: true
+      etcd:
+        local:
+          dataDir: /var/lib/etcddisk/etcd
+          extraArgs:
+            quota-backend-bytes: "8589934592"
+    diskSetup:
+      filesystems: []
+      partitions: []
+    files:
+    - contentFrom:
+        secret:
+          key: control-plane-azure.json
+          name: ${CLUSTER_NAME}-control-plane-azure-json
+      owner: root:root
+      path: /etc/kubernetes/azure.json
+      permissions: "0644"
+    format: ignition
+    ignition:
+      containerLinuxConfig:
+        additionalConfig: |
+          systemd:
+            units:
+            - name: kubeadm.service
+              dropins:
+              - name: 10-flatcar.conf
+                contents: |
+                  [Unit]
+                  After=oem-cloudinit.service
+          # Workaround for https://github.com/kubernetes-sigs/cluster-api/issues/7679.
+          storage:
+            disks:
+            - device: /dev/disk/azure/scsi1/lun0
+              partitions:
+              - number: 1
+            filesystems:
+            - name: etcd_disk
+              mount:
+                device: /dev/disk/azure/scsi1/lun0
+                format: ext4
+                label: etcd_disk
+                options:
+                - -E
+                - lazy_itable_init=1,lazy_journal_init=1
+    initConfiguration:
+      nodeRegistration:
+        kubeletExtraArgs:
+          azure-container-registry-config: /etc/kubernetes/azure.json
+          cloud-config: /etc/kubernetes/azure.json
+          cloud-provider: azure
+        name: '@@HOSTNAME@@'
+    joinConfiguration:
+      nodeRegistration:
+        kubeletExtraArgs:
+          azure-container-registry-config: /etc/kubernetes/azure.json
+          cloud-config: /etc/kubernetes/azure.json
+          cloud-provider: azure
+        name: '@@HOSTNAME@@'
+    mounts:
+    - - etcd_disk
+      - /var/lib/etcddisk
+    postKubeadmCommands: []
+    preKubeadmCommands:
+    - sed -i "s/@@HOSTNAME@@/$(curl -s -H Metadata:true --noproxy '*' 'http://169.254.169.254/metadata/instance?api-version=2020-09-01'
+      | jq -r .compute.name)/g" /etc/kubeadm.yml
+  machineTemplate:
+    infrastructureRef:
+      apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+      kind: AzureMachineTemplate
+      name: ${CLUSTER_NAME}-control-plane
+  replicas: ${CONTROL_PLANE_MACHINE_COUNT}
+  version: ${KUBERNETES_VERSION}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-control-plane
+  namespace: default
+spec:
+  template:
+    spec:
+      dataDisks:
+      - diskSizeGB: 256
+        lun: 0
+        nameSuffix: etcddisk
+      image:
+        computeGallery:
+          gallery: flatcar4capi-742ef0cb-dcaa-4ecb-9cb0-bfd2e43dccc0
```

Comment:
would be good to add some docs around this and specifically the image. Are users meant to use this in production or is it just provided as an example? Is the image patched for security vulnerabilities?

Reply:
Right. The images will have initially no guarantees as other reference CAPI images, they are only for testing/demo purposes. We might consider changing that over time though.

We will work on the docs for it as well. If I knew they would be needed for this PR, I'd wait with opening it until it's done :smile:

Reply:
As soon as we expose something to users (in this case via the new flavor that will get published as a release asset) there should be docs for it. Otherwise, users might use it in a way that's unintended. We do have a documentation box on the PR description TODOs and it's documented in https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/CONTRIBUTING.md#contributing-a-patch

> All new features should come with user documentation.

Reply:
Hmm, do you have some suggestion where it should be documented? I'm looking at https://capz.sigs.k8s.io/topics/custom-images.html and https://capz.sigs.k8s.io/introduction.html?highlight=flavor#flavors but I don't immediately see any good place to put such docs. I'd assume this images should be treated no different from other images used by CAPZ, except that they will be delivered by a different team for the time being, until CAPZ itself migrate to use SIG images.

Reply:
We could start a new "Flatcar" topic page with an image section just like Windows https://capz.sigs.k8s.io/topics/windows.html#image-creation

Reply:
Sounds good, thanks. I'll prepare some draft next week.

Reply:
Added some initial docs so we can have something to comment on. Please have a look.